src/mpi/coll/ibarrier/ibarrier.c

```diff
@@ -166,22 +166,21 @@ int MPIR_Ibarrier_impl(MPIR_Comm * comm_ptr, MPIR_Request ** request)
                 break;
         }
     }
-    if (comm_ptr->local_size != 1 || comm_ptr->comm_kind == MPIR_COMM_KIND__INTERCOMM) {
```

Comment:
We solved this issue previously by allocating an already complete request before `MPI_Ibarrier` returns to the user. https://github.com/pmodels/mpich/blob/master/src/mpi/coll/ibarrier/ibarrier.c#L291

Was there some issue that wasn't covered by the current solution?

Reply:
If not, I think this patch can be dropped.